libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -500,6 +500,8 @@ int lua_get_CAN_device2(lua_State *L) {
   directory listing, return table of files in a directory
  */
 int lua_dirlist(lua_State *L) {
+    binding_argcheck(L, 1);
```

Comment:
why this change?

Reply:
Checks you actually pass 1 argument, so `dirlist('foo', 10)` will now give a error. previously it would have just ignored the extra arguments. Also I'm not exactly sure what `dirlist()` would do now with no arguments. It might try and convert the next thing on the stack to a string. 

The arg check will error out with either "too many arguments" or "too few arguments".

However your right its more of a drive by in this PR. I can split it off into a separate one if you prefer.

Reply:
`dirlist()` does fail correctly, but you can just pass extra arguments with no error, so this change stops that.